packages/grafana-ui/src/components/OptionsUI/number.tsx

```diff
@@ -13,18 +13,27 @@ export const NumberValueEditor: React.FC<FieldConfigEditorProps<number, NumberFi
   item,
 }) => {
   const { settings } = item;
+  const onValueChange = useCallback(
+    (value: string) => {
+      onChange(settings?.integer ? toIntegerOrUndefined(value) : toFloatOrUndefined(value));
+    },
+    [onChange]
+  );
   return (
     <Input
-      value={isNaN(value) ? '' : value}
+      defaultValue={isNaN(value) ? '' : value.toString()}
       min={settings?.min}
       max={settings?.max}
       type="number"
       step={settings?.step}
       placeholder={settings?.placeholder}
-      onChange={e => {
-        onChange(
-          settings?.integer ? toIntegerOrUndefined(e.currentTarget.value) : toFloatOrUndefined(e.currentTarget.value)
-        );
+      onBlur={e => {
+        onValueChange(e.currentTarget.value);
+      }}
+      onKeyDown={e => {
+        if (e.key === 'Enter') {
+          onValueChange(e.currentTarget.value);
+        }
```

Comment:
nit: maybe try to remove these anonymous functions, I guess the useCallback not so useful otherwise?

Reply:
Good catch! Primarily this is done to handle difference between form and keyboard events. But I think we can handle that with a generic event type and simply make sure `e.key && e.key === 'Enter'` to handle keyboard.